api/envoy/config/rbac/v3/rbac.proto

```diff
@@ -407,6 +416,10 @@ message Principal {
     // Matches against metadata from either dynamic state or route configuration. Preferred over the
     // ``metadata`` field as it provides more flexibility in metadata source selection.
     SourcedMetadata sourced_metadata = 13;
+
+    // Extension for configuring custom principals for RBAC.
+    // [#extension-category: envoy.rbac.principals]
+    core.v3.TypedExtensionConfig custom = 14;
```

Comment:
It looks like the implementation is using the same extension API as used in the [`matcher` field](https://github.com/envoyproxy/envoy/blob/65e9db7c5db4b4a1788158bad390223b08acca41/api/envoy/config/rbac/v3/rbac.proto#L303).  Would it make sense for this new extension to just plug in to that existing field, rather than adding a new one here?

One argument for keeping it here would be so that it's semantically clear that this is matching against the principle specifically.  I'm not sure how much that matters in practice.

I don't feel strongly about this either way -- just raising it as a question to consider.

Reply:
I'm trying to keep it consistent with principals/permissions as described in the docs (https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/rbac/v3/rbac.proto#envoy-v3-api-msg-config-rbac-v3-policy) and with the current mtls authentication config, so I think it makes more sense to make a new extension field here. cc @yangminzhu @yanavlasov as codeowners of RBAC extensions for input on this.